soc/arm/nxp_s32/s32k1/soc.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _NXP_S32_S32K1_SOC_H_
+#define _NXP_S32_S32K1_SOC_H_
+
+#include <fsl_port.h>
+
+#if defined(CONFIG_SOC_S32K116)
+#include <S32K116.h>
+#elif defined(CONFIG_SOC_S32K118)
+#include <S32K118.h>
+#elif defined(CONFIG_SOC_S32K142)
+#include <S32K142.h>
+#elif defined(CONFIG_SOC_S32K142W)
+#include <S32K142W.h>
+#elif defined(CONFIG_SOC_S32K144)
+#include <S32K144.h>
+#elif defined(CONFIG_SOC_S32K144W)
+#include <S32K144W.h>
+#elif defined(CONFIG_SOC_S32K146)
+#include <S32K146.h>
+#elif defined(CONFIG_SOC_S32K148)
+#include <S32K148.h>
+#else
+#error "SoC not supported"
+#endif
+
+#if defined(CONFIG_CMSIS_RTOS_V2)
```

Comment:
Pls rebase main, [this](https://github.com/zephyrproject-rtos/zephyr/pull/65716/files#diff-e78dbc7cb3c75660e74759ac8c451f762446b3585f8759d3b4c76119368c6f7d) was merged